lib/puppet/type/package.rb

```diff
@@ -437,12 +437,6 @@ def change_to_s(currentvalue, newvalue)
 
     newparam(:allow_virtual, :boolean => true, :parent => Puppet::Parameter::Boolean, :required_features => :virtual_packages) do
       desc 'Specifies if virtual package names are allowed for install and uninstall.'
-
-      # In a future release, this should be defaulted to true and the below deprecation warning removed
-      defaultto do
-        Puppet.deprecation_warning('The package type\'s allow_virtual parameter will be changing its default value from false to true in a future release. If you do not want to allow virtual packages, please explicitly set allow_virtual to false.') unless value
-        false
```

Comment:
For stable (3.x) I think we want to retain the default value of false here. The reason for the default of false was that we saw the `allow_virtual=true` behavior as making more sense long-term, but as a breaking change, so we were planning to not change behavior in the 3.x series. I think that logic still holds. But I may be missing sth since it's been a few months.

Assuming we do keep this as `false`, we do need to take care on the merge-up to master that we keep the default at `true` on master.